File: src/main/java/Solution_28_FindIndexOfFirstOccurrenceInString.java
public class Solution_28_FindIndexOfFirstOccurrenceInString {
    public static void main(String[] args) {
        var haystack = "sadbutsad";
        var needle = "sad";
        var result = strStr(haystack, needle);
        System.out.println(result);

    }

    private static int strStr(String haystack, String needle) {
        if (haystack.length() < needle.length()) return -1;
        if (haystack.equals(needle)) return 0;

        var hArr = haystack.toCharArray();
        var nArr = needle.toCharArray();
        var targetSize = 0;

        for (int i = 0; i < haystack.length(); i++) {
            for (int j = 0; j < haystack.length(); j++) {
                if (i + j == hArr.length || hArr[i + j] != nArr[j]) {
                    targetSize = 0;
                    break;
                }
                targetSize++;
                if (targetSize == needle.length()) return i;
            }
        }

        return -1;
    }
}

File: src/main/typescript/Solution_28_FindIndexOfFirstOccurrenceInString.ts
const haystack: string = "sadbutsad"
const needle: string = "sad"

console.log(strStr(haystack, needle))

function strStr(haystack: string, needle: string): number {
    if (haystack.length < needle.length) return -1
    if (haystack == needle) return 0

    for (let i = 0; i < haystack.length; i++) {
        let j = 0
        while (j < needle.length && haystack[i + j] === needle[j]) {
            j++;
        }
        if (j === needle.length) return i

    }

    return -1
}